stream/client.go

```diff
@@ -0,0 +1,83 @@
+package stream
+
+import (
+	"bytes"
+	"context"
+	"encoding/gob"
+	"fmt"
+	"net/http"
+
+	"github.com/google/zoekt"
+	"github.com/google/zoekt/query"
+	"github.com/google/zoekt/rpc"
+)
+
+type client struct {
+	address string
+	path    string
+	conn    *http.Client
+}
+
+func NewClientAtAddress(address string) *client {
+	rpc.RegisterGob()
+	return &client{address, DefaultSSEPath, &http.Client{
+		Transport: &http.Transport{
+			MaxIdleConns: 500,
+		},
+	}}
```

Comment:
why the hardcoded client? I would expect a user to pass it in so it can be shared and have other values set on it.

How about something like

```go
type Client struct {
  // HTTP address of zoekt-webserver. Will query against Address + "/stream".
  Address string

  // HTTPClient when set is used instead of http.DefaultClient
  HTTPClient *http.Client
}
```

if you take my suggestion, we just need to remember to call RegisterGob in StreamSearch.

Reply:
Sounds good.